syntax = "proto3";


enum UpdateType {
  UpdateType_NotSet = 0;
  UpdateType_None = 1;
  UpdateType_Interval = 2;
}

message UpdateConfiguration {
  UpdateType updateType = 1;
  uint32 updateIntervalInMs = 2;
}

message SetUpdateConfiguration {
  UpdateConfiguration updateConfiguration = 1;
  bool saveConfiguration = 2;
}

message HatConfiguration {
  bool rotateImage = 1;
}

message SetHatConfiguration {
  HatConfiguration hatConfiguration = 1;
  bool saveConfiguration = 2;
}

enum NoArgCommands {
  Nac_NotSet             = 0;
  Nac_ClearScreenToWhite = 1;
  Nac_ClearScreenToBlack = 2;
  Nac_GetHatDetails      = 3;
  Nac_GetFirmwareDetails = 4;
  Nac_RestartDevice      = 5;
}

message MyProjectCommand {
  oneof command {
    NoArgCommands noArgCommand = 1;
    string showFileImage = 2;
    SetUpdateConfiguration setUpdateConfiguration = 3;
    SetHatConfiguration setHatConfiguration = 4;
  }
}



message InvalidCommandResponse {
  string message = 1;
}

message HatDetailsResponse {
  UpdateConfiguration updateConfiguration = 1;
  HatConfiguration hatConfiguration = 2;
  string activeImageFile = 3;
  string activeImageLabel = 4;
}

message FirmwareDetailsResponse {
  string version = 1;
  string buildTime = 2;
}

message UpdateResponse {
  uint32 boardUptimeInMs = 1;
}

message MyProjectResponse {
  oneof response {
    InvalidCommandResponse invalidCommandResponse = 1;
    HatDetailsResponse hatDetailsResponse = 2;
    FirmwareDetailsResponse firmwareDetailsResponse = 3;
    UpdateResponse updateResponse = 4;
  }
}
